Cleanup Kconfig from board DTS

nashif opened the pull request:
Move to overlays and remove configurable features from boards.

nashif committed:
boards: dts: define default console only

When using USB CDC, set that via an overlay instead of in the board DTS.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: make ethernet build test depend on netif

Test for ethernet should depend on the HW having ethernet support.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

finikorg commented on boards/x86/tinytile/tinytile.dts:
```diff
@@ -20,11 +20,7 @@
-#else
 		zephyr,console = &uart1;
-#endif
 		zephyr,bt-uart = &uart0;
```
Does the board has only USB Serial console only? How the board works after that?

nashif replied:
> Does the board has only USB Serial console only?

no, it does have UART.

Board will work exactly the same way, if you want to use CDC, then you will need an overlay.

finikorg commented on samples/subsys/usb/console/tinytile.overlay:
```diff
@@ -0,0 +1,5 @@
+        chosen {
+                zephyr,console = &usb_cdc;
+        };
+};
```
How to include these overlays to build?

nashif replied:
this is included by the build system directly.

galak reviewed (dismissed):
If you are removing CONFIG_BT, we should remove CONFIG_MODEM_WNCM14A2A

nashif commented:
> If you are removing CONFIG_BT, we should remove CONFIG_MODEM_WNCM14A2A

will come in another PR, there is no dependency between the two.

nashif committed:
boards: frdm_k64f: move modem configurtion to overlay

The wncm14a2a modem is an add-on/shield and not part of the frdm_k64f
board and should be maintained and configured outside of the board.

This needs to be moved to a shield, see #10965.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
> If you are removing CONFIG_BT, we should remove CONFIG_MODEM_WNCM14A2A

ok, added a commit now, but this should end up in a shield IMO. Filed issue #10965.

galak commented:
Looks good to me, want to let it sit around for a bit for @mike-scott and @MaureenHelm to have a chance to comment.

finikorg commented on boards/arm/frdm_k64f/dts_fixup.h:
```diff
@@ -13,19 +13,3 @@
 #define CONFIG_MCR20A_GPIO_IRQ_B_PIN			NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_IRQB_GPIOS_PIN
 #define CONFIG_MCR20A_GPIO_RESET_NAME			NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_RESET_GPIOS_CONTROLLER
 #define CONFIG_MCR20A_GPIO_RESET_PIN			NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_RESET_GPIOS_PIN
```
Should we do the same for MCR20A?

MaureenHelm reviewed (changes requested):
@nashif I guess you're all in on dts overlays now? 😄 I'll look at adding a bt shield overlay for kw41

nashif commented on boards/arm/hexiwear_k64/hexiwear_k64.dts:
```diff
@@ -37,9 +37,6 @@
 		zephyr,flash = &flash0;
 		zephyr,console = &uart0;
-#ifdef CONFIG_BT
-		zephyr,bt-uart = &uart4;
```
ok, we can just remove the ifdef then.

nashif replied:
leaving that was causing build issues, because uart4 is not defined?

nashif commented:
> @nashif I guess you're all in on dts overlays now?

not really, but we have to move forward, we will find a way to make them user-friendly soon enough :)

mike-scott commented:
@nashif I'm ok with the modem removal here.  I'll set it up as a shield in a new PR shortly.

galak commented on boards/arm/frdm_k64f/frdm_k64f.dts:
```diff
@@ -119,33 +116,10 @@
 };
 
-#ifdef CONFIG_UART_MCUX_2
 &uart2 {
```
uart2 should be removed here, this was done just for the Modem

nashif committed:
boards: dts: remove CONFIG_BT checking

No need to check for BT being enabled. The UART entries will be
available for use if BT is enabled and will do nothing if not.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented on tests/drivers/build_all/testcase.yaml:
```diff
@@ -13,7 +13,7 @@ tests:
     platform_exclude: zedboard_pulpino frdm_kw41z
     tags: drivers footprint
-    depends_on: adc spi
+    depends_on: adc spi netif
```
fixed

MaureenHelm commented on boards/arm/hexiwear_k64/hexiwear_k64.dts:
```diff
@@ -37,9 +37,7 @@
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
 		zephyr,console = &uart0;
-#ifdef CONFIG_BT
```
The changes here are fine, but the commit message doesn't make sense anymore. Will you please update the commit message?
